spring-cloud-stream-test-support/src/main/java/org/springframework/cloud/stream/test/binder/TestSupportBinder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.test.binder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.BlockingQueue;
+
+import org.springframework.cloud.stream.binder.Binder;
+import org.springframework.cloud.stream.test.matcher.MessageQueueMatcher;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+
+/**
+ * A minimal binder that<ul>
+ *     <li>does nothing about binding consumers, leaving the channel as-is, so that a test author can interact with it directly,</li>
+ *     <li>registers a queue channel on the producer side, so that it is easy to assert what is received.</li>
+ * </ul>
+ *
+ * @author Eric Bottard
+ * @see MessageQueueMatcher
+ */
+public class TestSupportBinder implements Binder<MessageChannel> {
+
+	public static Map<MessageChannel, QueueChannel> channelToBindings =
+			new HashMap<>();
+
+	private final MessageCollector messageCollector;
+
+	public TestSupportBinder(MessageCollector messageCollector) {
+		this.messageCollector = messageCollector;
+	}
+
+
+	@Override
+	public void bindConsumer(String name, MessageChannel inboundBindTarget, Properties properties) {
+		// Do nothing. A module author can grab hold forChannel the input channel
```

Comment:
Hmm, weird sentence indeed. May be IDEA refactoring from text in comments. The idea is already expressed in the class javadoc anyhow. Will remove